src/components/legend.ts

```diff
@@ -144,13 +164,17 @@ export module Components {
 
       let availableWidthForEntries = Math.max(0, (availableWidth - this._padding));
 
-      let entryNames = this._colorScale.domain().slice();
-      entryNames.sort(this.comparator());
+      let entryLabels: d3.Map<string> = d3.map<string>();
+      this._colorScale.domain().slice().forEach((d) => {
+        entryLabels.set(this._formatter(d), d);
+      });
+      let entryNames = entryLabels.keys();
+      entryNames = entryNames.sort(this.comparator()).map((d) => entryLabels.get(d));
```

Comment:
The variable names here can be quite a bit confusing.

It seems that `entryNames` is trying to be a variable for 2 different concepts.  This can be bad, because if one of the 2 concepts needs to change, then the variable here has to be drastically modified.  One is that it holds the formatted transform of the colorScale's domain.  Two is that is also eventually holds the original colorScale's domain in sorted form.

Can we separate these two variables out and see how it reads?